apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -146,20 +146,28 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
   }
 
   /**
-   * Serializes the PNPM Shrinkwrap file
+   * Gets the path to the tarball file if the package is a tarball,
+   * otherwise returns undefined. Example: file:projects/build-tools.tgz
    */
-  protected serialize(): string {
-    return yaml.safeDump(this._shrinkwrapJson, SHRINKWRAP_YAML_FORMAT);
+  public getTarballPath(packageName: string): string | undefined {
+    return this._shrinkwrapJson.packages[packageName].resolution.tarball;
```

Comment:
We should have a defensive check around `this._shrinkwrapJson.packages[packageName]`. It should probably return `undefined` if `this._shrinkwrapJson.packages[packageName]` is `undefined` instead of throwing.

Reply:
Hmm, makes sense.